project/MavenPluginIntegration.scala

```diff
@@ -0,0 +1,298 @@
+package ch.epfl.scala.sbt.maven
+
+import java.nio.charset.Charset
+
+import org.apache.maven.artifact.handler.DefaultArtifactHandler
+import org.apache.maven.artifact.{Artifact, DefaultArtifact}
+import org.apache.maven.model.Build
+import org.apache.maven.plugin.descriptor.{MojoDescriptor, Parameter, PluginDescriptor, PluginDescriptorBuilder, Requirement}
+import org.apache.maven.plugin.logging.SystemStreamLog
+import org.apache.maven.project.MavenProject
+import org.apache.maven.tools.plugin.DefaultPluginToolsRequest
+import org.codehaus.plexus.logging.Logger
+import org.codehaus.plexus.logging.console.ConsoleLogger
+import org.apache.maven.tools.plugin.extractor.annotations.scanner.{DefaultMojoAnnotationsScanner, MojoAnnotationsScannerRequest}
+import org.apache.maven.tools.plugin.generator.PluginDescriptorGenerator
+import sbt.{AutoPlugin, Compile, Def, Keys, PluginTrigger, Plugins}
+import org.codehaus.plexus.component.repository.{ComponentDependency, ComponentRequirement}
+import sbt.librarymanagement.UpdateReport
+
+object MavenPluginIntegration extends AutoPlugin {
+  override def trigger: PluginTrigger = allRequirements
+  override def requires: Plugins = sbt.plugins.JvmPlugin
+  val autoImport = MavenPluginKeys
+
+  override def globalSettings: Seq[Def.Setting[_]] =
+    MavenPluginImplementation.globalSettings
+  override def buildSettings: Seq[Def.Setting[_]] =
+    MavenPluginImplementation.buildSettings
+  override def projectSettings: Seq[Def.Setting[_]] =
+    MavenPluginImplementation.projectSettings
+}
+
+object MavenPluginKeys {
+  import sbt.{settingKey, taskKey}
+  val mavenPlugin = settingKey[Boolean](
+    "If true, enables adding maven as a dependency and auto-generation of the plugin descriptor file.")
+  val mavenLogger = settingKey[Logger]("The Plexus logger used for Maven APIs.")
+  val mavenProject =
+    taskKey[MavenProject]("Return the corresponding maven project.")
```

Comment:
`s/corresponding/current/`?

Reply:
It doesn't have quite the same meaning, why do you prefer `current`?

Reply:
Because otherwise it doesn't say to what it corresponds. `corresponding to the current project` then?

Reply:
yes, i'll do that :smile: